index.js

```diff
@@ -372,60 +433,300 @@ export const sendForm = method => url => headers => form => {
+//. A redirection strategy that simply reissues the original Request to the
+//. Location specified in the given Response. If the response does not contain
+//. a valid location, the request is not redirected.
+//.
+//. Used in [`followRedirects`](#followRedirects) and
+//. [`followRedirectsStrict`](#followRedirectsStrict).
+export const redirectOriginalRequest = response => {
+  const {headers: {location}} = Response.message (response);
+  const original = Response.request (response);
+  const request = Request (Request.options (original))
+                          (location)
+                          (Request.body (original));
+  return (
+    isValidUrl (location) ?
+    mapRej (prefixErrorMessageRedirect) (sendRequest (request)) :
+    resolve (response)
+  );
+};
+
+//# redirectIfGetMethod :: Response -> Future Error Response
+//.
+//. A redirection strategy that simply reissues the original Request to the
+//. Location specified in the given Response, but only if the original request
+//. was using the GET method.
+//.
+//. Used in [`followRedirectsStrict`](#followRedirectsStrict).
+export const redirectIfGetMethod = response => {
+  const {method} = Request.options (Response.request (response));
+  return (
+    typeof method === 'string' && method.toUpperCase () !== 'GET' ?
+    resolve (response) :
+    redirectOriginalRequest (response)
+  );
+};
+
+//# redirectUsingGetMethod :: Response -> Future Error Response
+//.
+//. A redirection strategy that sends a new GET request based on the original
+//. request to the Location specified in the given Response. If the response
+//. does not contain a valid location, the request is not redirected.
+//.
+//. The original request method and body are discarded, but all the options
+//. are preserved.
+//.
+//. Used in [`followRedirects`](#followRedirects) and
+//. [`followRedirectsStrict`](#followRedirectsStrict).
+export const redirectUsingGetMethod = response => {
+  const {headers: {location}} = Response.message (response);
+  const original = Response.request (response);
+  const options = Object.assign ({}, Request.options (original), {
+    method: 'GET',
+  });
+  const request = Request (options) (location) (emptyStream);
+  return (
+    isValidUrl (location) ?
+    mapRej (prefixErrorMessageRedirect) (sendRequest (request)) :
+    resolve (response)
+  );
+};
+
+// TODO: Expand to include all headers that could cause servers to respond 304
+const conditionHeaders = ['etag'];
+
+//# retryWithoutCondition :: Response -> Future Error Response
+//.
+//. A redirection strategy that removes any caching headers if present and
+//. retries the request, or does nothing if no caching headers were present
+//. on the original request.
+//.
+//. Used in [`followRedirects`](#followRedirects).
+export const retryWithoutCondition = response => {
+  const original = Response.request (response);
+  const options = Request.options (original);
+  const headers = Object.entries (options.headers || {});
```

Comment:
I look forward to getting to use [`??`][1] one day. :)


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
Mee too!